modules/engine/src/transform/texture-transform.ts

```diff
@@ -0,0 +1,183 @@
+      ...props
+    });
+
+    this._initialize(props);
+    Object.seal(this);
+  }
+
+  // Delete owned resources.
+  destroy(): void {
+    if (this.elementIDBuffer) {
+      this.elementIDBuffer.destroy();
+    }
+  }
+
+  /** @deprecated Use {@link destroy}. */
+  delete(): void {
+    this.destroy();
+  }
+
+  run(options?: RenderPassProps): void {
+    const {framebuffer} = this.bindings[this.currentIndex];
+    const renderPass = this.device.beginRenderPass({framebuffer, ...options});
+    this.model.draw(renderPass);
+    renderPass.end();
+  }
+
+  /** @deprecated */
+  update(...args: any[]): void {
+    // TODO(v9): Method should likely be removed for v9. Keeping a method stub
+    // to assist with migrating DeckGL usage.
+    // eslint-disable-next-line no-console
+    console.warn('TextureTransform#update() not implemented');
+  }
+
+  getData({packed = false} = {}) {
+    throw new Error('getData() not implemented');
+  }
+
+  getTargetTexture(): Texture {
+    const {targetTexture} = this.bindings[this.currentIndex];
+    return targetTexture;
+  }
+
+
+  getFramebuffer(): Framebuffer {
+    const currentResources = this.bindings[this.currentIndex];
+    return currentResources.framebuffer;
+  }
+
+  // Private
+
+  _initialize(props: TextureTransformProps): void {
+    this._updateBindings(props);
+  }
+
+  _updateBindings(props: TextureTransformProps) {
+    this.bindings[this.currentIndex] = this._updateBinding(this.bindings[this.currentIndex], props);
+  }
+
+  _updateBinding(
+    binding: TextureBinding,
+    {sourceBuffers, sourceTextures, targetTexture}: TextureTransformProps
+  ): TextureBinding {
+    if (!binding) {
+      binding = {
+        sourceBuffers: {},
+        sourceTextures: {},
+        targetTexture: null
+      };
+    }
+    Object.assign(binding.sourceTextures, sourceTextures);
+    Object.assign(binding.sourceBuffers, sourceBuffers);
+    if (targetTexture) {
+      binding.targetTexture = targetTexture;
+      const {width, height} = targetTexture;
+      // TODO(donmccurdy): When is this called, and is this expected?
+      if (binding.framebuffer) {
+        binding.framebuffer.destroy();
+      }
+      binding.framebuffer = this.device.createFramebuffer({
+        id: 'transform-framebuffer',
+        width,
+        height,
+        colorAttachments: [targetTexture]
+      });
+      binding.framebuffer.resize({width, height});
+    }
+    return binding;
+  }
+
+  // set texture filtering parameters on source textures.
+  _setSourceTextureParameters(): void {
+    const index = this.currentIndex;
+    const {sourceTextures} = this.bindings[index];
+    for (const name in sourceTextures) {
+      sourceTextures[name].sampler = this.sampler;
+    }
+  }
+}
+
+/** Creates an element ID buffer, using float32 for GLSL 1.00 compatibility. */
+function createElementIDBuffer(device: Device, count: number): Buffer {
+  const data = new Float32Array(count);
```

Comment:
Can this not be done with `Uint16Array`?

Reply:
Appears this was for GLSL 1.0. Given other changes to luma v9 and other feedback in this PR, let's use `gl_InstanceID` yes.